Pulls/cluster

benmanu opened the pull request:
Adding clustering config option through the Leaflet.markercluster module.

digitalsadhu commented:
Firstly, can I say, thanks a tonne. This is awesome!

Secondly, @markstuart and I had a chat and have a suggestion for a minor change to the config API

The following should result in the clusterer being bypassed
```js
cluster: undefined 
cluster: null
cluster: false
```

The following should result in the cluster being activated

```js
cluster: true // results in layer = new L.MarkerClusterGroup();
```

```js
cluster: {} // results in layer = new L.MarkerClusterGroup({});
```

How does that sound @benmanu?

markstuart commented:
Yep, nice work @benmanu. It'd be great to see a little more documentation of the clustering feature in the README to if possible, even if it was just to point out the the options to pass in there were to override the default options and a link to the MarkerClusterer documentation.

Something along those lines would be awesome.

benmanu commented:
Thanks for the feedback @digitalsadhu and @markstuart. 

Regarding using npm to manage Leaflet.markercluster, that looks like a much better solution but it requires rebuilding the markercluster source using a different branch currently not tracked through npm (outlined [here](https://github.com/Leaflet/Leaflet.markercluster/issues/338)) to get it to play nice with Leaflet 0.8. If you have some ideas around a good way to include it I can dig a bit deeper into implementing it?

I'll chuck in the changes to defining clustering on layer config and will add to the readme, should be able to push that through shortly.

digitalsadhu commented:
I would say we can either:
- add some kind of post install hook to build the module
- fork the repo temporarily and refer to that until 0.8 is supported
- leave the vendor approach in place until 0.8 is supported

I think I would just go with what you've got for now and update to a dependency at a later stage

markstuart commented:
Seconded :)

Just a minor note on contribution here @benmanu, could you please just put a comment on here when you've made updates to a PR? We don't get notified when you push commits to existing PR's, so we may not see fixes/changes you make for a while. If you say 'Updated Readme' or whatever in here, we'll get notified, and can get stuff merged in faster.

Cheers :)

benmanu commented:
Cheers @markstuart, noted down. On that note, I've updated the Readme, and added another check to building the L.MarkerClusterGroup (will fail on 'undefined', 'null', or 'false', will pass on 'true', or '{}').

benmanu commented:
Cheers @markstuart, noted down. On that note, I've updated the Readme, and added another check to building the L.MarkerClusterGroup (will fail on 'undefined', 'null', or 'false', will pass on 'true', or '{}').

digitalsadhu commented:
:+1: I'm going to rebase this, do some last checks and then merge the smack out it!

markstuart commented on wrappers/map.js:
```diff
@@ -52,8 +54,17 @@ class Map extends EventEmitter {
+    // if clustering is defined then add a marker cluster layer
+    // else add a geoJson layer
+    if (typeof config.cluster !== 'undefined' && config.cluster) {
+      layer = new L.MarkerClusterGroup(config.cluster);
```
Does this work if `config.cluster === true`? `true` is valid to pass into the constructor?

digitalsadhu replied:
Seems to work as expected with true, false, undefined, null and {}

I'm not getting icons for the clusters though so looking into that now

digitalsadhu commented:
@benmanu Are you getting cluster icons showing?
I'm just getting numbers 

![map](https://www.dropbox.com/s/bkutkg89r4akpvr/Screenshot%202015-01-21%2018.29.37.png?dl=0)

markstuart commented:
I think you might have to style up the elements that have the numbers in them with a background image?

digitalsadhu commented:
My understanding looking at the library is that there is a default cluster style though?

On Wed, Jan 21, 2015 at 6:35 PM, Mark Stuart <notifications@github.com>
wrote:

> I think you might have to style up the elements that have the numbers in them with a background image?
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/mediasuitenz/mappy/pull/75#issuecomment-70787088

markstuart commented:
Is there css that isn't being included?

markstuart commented:
The npm module readme mention a Default.css file

digitalsadhu commented:
sounds like a lead

digitalsadhu commented:
Merged in #77 (had to rebase and fix a couple things that broke as well as adjusted the README slightly) 

Thanks @benmanu and co!!
